src/polyline.js

```diff
@@ -134,21 +134,41 @@ polyline.fromGeoJSON = function(geojson, precision) {
     if (geojson && geojson.type === 'Feature') {
         geojson = geojson.geometry;
     }
-    if (!geojson || geojson.type !== 'LineString') {
-        throw new Error('Input must be a GeoJSON LineString');
+
+    if (!geojson) {
+        throw new Error('Input must be a valid GeoJSON');
+    }
+
+    var coords;
+    if (geojson.type === 'Point') {
+        coords = [geojson.coordinates];
+    } else if (geojson.type === 'LineString') {
+        coords = geojson.coordinates;
+    } else if (geojson.type === 'Polygon') {
+        // Any holes (coords index > 0) will be ignored
+        coords = geojson.coordinates[0];
```

Comment:
For a comprehensive Polygon support, we could encode the rings individually and join the strings using `\n`

Alternatively, the output could be an array of polyline strings